### PR TITLE
Add fish completion

### PR DESCRIPTION
**Pull request by xfgusta**
Here are some things I couldn't figure out how to implement:

```text
!
(
)
,
-newerXY REFERENCE
```

I also don't know if I should add a completion for `-noleaf`.

The completion is installed in the [`/usr/share/fish/vendor_completions.d` directory](https://fishshell.com/docs/current/completions.html#where-to-put-completions).

See #32.

### CODE REVIEW

**Comment by xfgusta** [comment]
> These are handled by expanding out the full set of possibilities for bash and zsh. Maybe fish can do something similar?

I managed to do this using the following code:

```fish
# handle -newer{a,B,c,m}{a,B,c,m} FILE
for x in {a,B,c,m}
    for y in {a,B,c,m}
        complete -c bfs -o newer$x$y -d "Find files whose $x""time is newer than the $y""time of specified file" -F
    end
end

# handle -newer{a,B,c,m}t TIMESTAMP
for x in {a,B,c,m}
    complete -c bfs -o newer$x"t" -d "Find files whose $x""time is newer than specified timestamp" -x
end
```
It generates these completions (except the first one):

image

Does it look right?

> The `bash` and `zsh` ones have it, so `fish` might as well too.

Added.

**Comment by tavianator** [comment]
Looks right to me.  `$x""time` looks weird but I guess it's right, since `fish` doesn't support `${x}time`.

**Comment by xfgusta** [comment]
> `$x""time` looks weird but I guess it's right, since `fish` doesn't support `${x}time`.

Unfortunately I couldn't come up with a better solution. Maybe something like this:

```fish
string join '' "Find files whose " $x "time is newer than the " $y "time of specified file"
```

**Comment by tavianator** [comment]
I think `$x""time` is better than that, might as well keep it

File: completions/bfs.fish [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+complete -c bfs -o exec -d "Execute a command" -r
+complete -c bfs -o ok -d "Prompt the user whether to execute a command" -r
+complete -c bfs -o execdir -d "Like -exec, but run the command in the same directory as the found file(s)" -r
+complete -c bfs -o okdir -d "Like -ok, but run the command in the same directory as the found file(s)" -r
+complete -c bfs -o exit -d "Exit immediately with the given status" -x
+complete -c bfs -o fls -d "Like -ls, but write to specified file" -F
+complete -c bfs -o fprint -d "Like -print, but write to specified file" -F
+complete -c bfs -o fprint0 -d "Like -print0, but write to specified file" -F
+complete -c bfs -o fprintf -d "Like -printf, but write to specified file" -F
+complete -c bfs -o ls -d "List files like ls -dils" -r
```
Comment (tavianator) [inline completions/bfs.fish:136]: Should this have `-r` at the end?  It looks like `-r` is `--require-parameter`, but `-ls` takes no parameters.

**Review by tavianator — COMMENTED** [review]
Looks good!  I tried it out locally and it works.  Just have the one question about `-ls`.

File: completions/bfs.fish (hunk 1 above)
Comment (xfgusta) [reply completions/bfs.fish:136]: It shouldn't. I will fix it.

**Pull request merged**